templatefuncs.go

```diff
@@ -159,6 +175,62 @@ func regexpReplaceAllTemplateFunc(expr, repl, s string) string {
 	return regexp.MustCompile(expr).ReplaceAllString(s, repl)
 }
 
+// reverseTemplateFunc is the core implementation of the `reverse`
+// template function.
+func reverseTemplateFunc(list []any) []any {
+	listcopy := append([]any(nil), list...)
+	slices.Reverse(listcopy)
+	return listcopy
+}
+
+// sortTemplateFunc is the core implementation of the `sort` template function.
+func sortTemplateFunc(list []any) any {
+	if len(list) < 2 {
+		return list
+	}
+
+	firstElemType := reflect.TypeOf(list[0])
+
+	for _, elem := range list[1:] {
+		if reflect.TypeOf(elem) != firstElemType {
+			return list
+		}
+	}
+
+	switch firstElemType.Kind() { //nolint:exhaustive
```

Comment:
No need to be exhaustive here. It's sufficient to cover the common types (`int`, `float64`, `string`, and `bool`).

Reply:
If I don't disable `exhaustive` the linter complains because I'm not accounting for the other `Kind` cases:

```
missing cases in switch of type reflect.Kind: reflect.Invalid, reflect.Bool, reflect.Complex64, reflect.Complex128, reflect.Array, reflect.Chan, reflect.Func, reflect.Interface, reflect.Map, reflect.Pointer|reflect.Ptr, reflect.Slice, reflect.Struct, reflect.UnsafePointer (exhaustive)
```

Reply:
You can use a `default:` case which `panic`s with an "unsupported type" message.

Reply:
I'm already using a `default` case, though.

Granted, it doesn't `panic`, but I just changed it and it also doesn't satisfy the linter:

![image](https://github.com/chezmoi/templatefuncs/assets/23157013/e23c0f49-ecbe-4470-9a12-74485f0a8515)

Reply:
Hmm, in that case feel free to disregard/disable the linter. Note that this exhaustive checking might be covered by multiple linters and you might need to disable all of them. Also, check to see if the warning is coming from golangci-lint with chezmoi's configuration, or if it's an extra linter being run by your IDE.

Reply:
https://pkg.go.dev/github.com/nishanths/exhaustive#hdr-Definition_of_exhaustiveness

We can add the `-default-signifies-exhaustive` flag to the linter if you'd prefer.

Reply:
Also, I'm using `golangci-lint` as the linter in VS Code with [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens), as well as [Run on Save](https://marketplace.visualstudio.com/items?itemName=emeraldwalk.RunOnSave) to format:

```json
{
  "go.lintTool": "golangci-lint",
  "go.lintFlags": ["--fast"],
  "emeraldwalk.runonsave": {
    "commands": [
      {
        "match": "\\.go$",
        "cmd": "gci write ${file} --skip-generated -s standard -s default -s prefix(github.com/chezmoi/templatefuncs)"
      },
      {
        "match": "\\.go$",
        "cmd": "golines --base-formatter=\"gofumpt -extra\" --max-len=128 --write-output ${file}"
      }
    ]
  }
}
```